OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -18,6 +18,28 @@ namespace OpenRA.Mods.Common
 {
 	public static class ActorExts
 	{
+		public static bool IsAtGroundLevel(this Actor self)
+		{
+			if (self.IsDead)
+				return false;
+
+			if (!self.HasTrait<IPositionable>())
+				return false;
+
+			if (!self.IsInWorld)
+				return false;
+
+			var map = self.World.Map;
+
+			if (!map.Contains(self.Location))
+				return false;
+
+			var centerPos = self.CenterPosition;
```

Comment:
You aren't doing anything interesting with these temporary variables, so just `return map.DistanceAboveTerrain(self.CenterPosition) == 0`.